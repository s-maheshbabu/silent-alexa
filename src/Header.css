#header {
  grid-row: 1;
  grid-column: 1 / 3;

  display: grid;
  grid-template-columns: 1fr auto;
}

#header-info {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;

  display: flex;
  justify-content: center;
}

#header-controls {
  display: flex;
  height: 100%;
  justify-content: flex-end;
}