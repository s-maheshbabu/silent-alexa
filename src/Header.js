import React from "react";
import LoginControl from "./LoginControl.js";

export default function Header(props) {
  return (
    <div id="header">
      {/* TODO: Currently the content in header-info isn't centered to the entire page but is at the center of header-info itself. 
              It should be adjust to be at the center of the entire page. */}
      <div id="header-info">
        <span>Silent Alexa Header</span>
      </div>
      <div id="header-controls">
        <LoginControl updateAuthenticationInfo={(authResponse) => props.updateAuthenticationInfo(authResponse)} />
      </div>
    </div>
  );
}