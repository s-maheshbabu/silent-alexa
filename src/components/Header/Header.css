/** Scaffolding **/
#header {
  grid-row: 1;
  grid-column: 1 / 3;

  display: grid;
  grid-template-columns: 1fr auto;
  border-bottom: 5px solid;
  /** Alexa light ring effect */
  border-image: linear-gradient(
      to right,
      #5291de,
      #5291de 40%,
      #16eaed 42%,
      #16eaed,
      #16eaed 58%,
      #5291de 60%,
      #5291de
    )
    60;
  /** fallback, when border-image can't be used TODO TODO TODO */
  border-style: xxx;
}

#header-info {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;

  display: flex;
  justify-content: center;
}

.app-bar {
  background: #152939 !important;
}
