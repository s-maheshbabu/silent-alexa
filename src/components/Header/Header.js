import React from "react";
import "./Header.css";
import LoginControl from "LoginControl/LoginControl";
import AppBar from "material-ui/AppBar";
import MuiThemeProvider from "material-ui/styles/MuiThemeProvider";

/**
 * Exported to facilitate testing.
 */
function _LoginControlInHeader(props) {
  return (
    <LoginControl
      isAuthenticationInfoValid={() => props.isAuthenticationInfoValid()}
      clearAuthenticationInfo={() => props.clearAuthenticationInfo()}
      updateAuthenticationInfo={authenticationInfo =>
        props.updateAuthenticationInfo(authenticationInfo)
      }
    />
  );
}

export default function Header(props) {
  return (
    <div id="header">
      <div id="header-info">
        <MuiThemeProvider>
          <AppBar
            className="app-bar"
            showMenuIconButton={false}
            title="Silent Alexa (Under Construction)"
            iconElementRight={_LoginControlInHeader(props)}
          />
        </MuiThemeProvider>
      </div>
    </div>
  );
}

export { _LoginControlInHeader };
