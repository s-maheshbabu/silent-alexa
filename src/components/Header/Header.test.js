import React from "react";
import { shallow } from "enzyme";

import Header from "./Header";
it("renders Header without crashing", () => {
  const wrapper = shallow(<Header />);
  expect(wrapper).toMatchSnapshot();

  wrapper.unmount();
});
