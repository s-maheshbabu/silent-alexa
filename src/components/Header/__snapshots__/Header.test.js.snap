// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Header without crashing 1`] = `
<div
  id="header"
>
  <div
    id="header-info"
  >
    <MuiThemeProvider>
      <AppBar
        className="app-bar"
        iconElementRight={<LoginControl />}
        showMenuIconButton={false}
        title="Silent Alexa (Under Construction)"
        zDepth={1}
      />
    </MuiThemeProvider>
  </div>
</div>
`;
